src/plugins/intel_gpu/src/graph/gemm.cpp

```diff
@@ -181,13 +184,16 @@ std::vector<layout> gemm_inst::transform_input_layouts(const std::shared_ptr<con
 
     bool reordered = primitive->input_rank > 4 || primitive->weight_rank > 4;
     size_t output_rank = std::max(primitive->input_rank, primitive->weight_rank);
-    size_t input_rank = reordered ? output_rank : primitive->input_rank;
-    size_t weight_rank = reordered ? output_rank : primitive->weight_rank;
+    size_t input_rank = (reordered && (input0_pshape.size() > primitive->input_rank))  ? output_rank : primitive->input_rank;
+    size_t weight_rank = (reordered && (input1_pshape.size() > primitive->weight_rank)) ? output_rank : primitive->weight_rank;
```

Comment:
is `reordered` required in such code change? Let's discuss offline.

Reply:
I changed condition here to check allow_new_shape_infer of program. new_shape_infer=true case doesn't need to fallback to output rank here.
But new_shape_infer=false case needs this for rank>4 case. It affects to transpose order.